### ISSUE TITLE
Reacting to Microsoft.Framework -> Microsoft.Extension renames

### ISSUE THREAD

**Issue by pranavkm** [autofac/Autofac#685]
We made some package \ namespace renames in Aspnet5 as part of an API review: https://github.com/aspnet/Announcements/issues/77.

The renames affected the DependencyInjection.Abstractions package and consequently broke our ability to consume `Autofac.Framework.DependencyInjection` in our tests

#### Before:
**Package**: Microsoft.Framework.DependencyInjection
**Fully qualified type name**: `Microsoft.Framework.DependencyInjection.ServiceDescriptor`

## After:
**Package**: Microsoft.Extensions.DependencyInjection
**Fully qualified type name**: `Microsoft.Extensions.DependencyInjection.ServiceDescriptor`

We should have the updated packages on the aspnetvnext feed soon. Using this as tracking item so we can revive the Autofac tests once you push a compatible `Autofac.Framework.DependencyInjection` \ `Autofac.Extensions.DependencyInjection` package to https://www.myget.org/F/autofac/api/v2/

**Comment by tillig** [comment]
I'm on it, though it may be tomorrow before I get there. We'll probably rename to keep in line with the new naming convention.

**Comment by tillig** [comment]
Note: [this is a change for RC1](https://github.com/aspnet/DependencyInjection/issues/297).

**Comment by tillig** [comment]
The `Autofac.Framework.DependencyInjection` package has been renamed to `Autofac.Extensions.DependencyInjection` and a DNX beta8-15616 compatible version is on the Autofac MyGet feed. The `Microsoft.Extensions.DependencyInjection` library isn't available quite yet so that dependency isn't updated, but the Autofac naming is in line. When the updated dependency is available, we can switch over.

Since it's an RC1 change, it'd be nice to release an "official" beta 8 compatible build before switching over, but beta 8 appears to have been pushed by a week so we'll have to watch the timing. This would be potentially easier [if the release version of the betas were named differently](https://github.com/aspnet/Home/issues/890).

**Comment by pranavkm** [comment]
@tillig thanks. We had an test failure unrelated to the naming change over the weekend. Still waiting on it to get resolved before the packages get uploaded to aspnetvnext.

**Comment by jruckert** [comment]
@tillig any luck on updating the packages to your MyGet stream with this change?

**Comment by tillig** [comment]
No status change to report. Don't worry, we'll post in here when it's done.

**Comment by pranavkm** [comment]
Our packages are up on MyGet. This change targets RC1 (so it's no `aspnetvnext`). For the issue you filed - https://github.com/aspnet/Home/issues/890 - we're planning on going with your suggestion. Unfortunately the beta8 milestone is largely locked down so we won't be able to target this change for the current milestone.

**Comment by tillig** [comment]
:+1: That will make things way easier since I won't have to wait until "real" beta 8 final release to build our beta 8 final.

I'll see about getting an Autofac beta 8 "final" ready on MyGet and then switch the naming for rc1 to the new dependencies. Not sure if it'll be today; "the day job" has struck pretty hard.

**Comment by tillig** [comment]
Can we get an update on https://github.com/aspnet/Home/issues/890 if you're actually going that way?

**Comment by tillig** [comment]
It looks like the beta 8 final will still be -beta8 and not -beta8-final so I probably won't push updated Autofac packages with the new dependencies until after beta 8 is actually released. If I can't create an "early final" for beta 8, it means I'd start messing with branches and setting up for rc1 whilst still waiting for final beta 8, etc. Given my current time constraints, it makes more sense to just wait for the beta 8 drop.

...unless you can get beta 8 to use the -beta8-final style of naming... :)

**Comment by pranavkm** [comment]
@tillig unfortunately I can't make that call. I think it would be fine to wait for beta8 to be released.

**Comment by dneimke** [comment]
Hi @tillig, any idea on a timeframe for releasing a build which compiles against the Microsoft.Extensions.DependencyInjection library?

E.g. the following file still has a using statement for Microsoft.Framework.DependencyInjection and breaks when targeting the latest Microsoft ASP.NET libraries

https://github.com/autofac/Autofac/blob/develop/src/Autofac.Extensions.DependencyInjection/AutofacRegistration.cs

**Comment by tillig** [comment]
@dneimke See above - waiting for beta 8 drop.

**Comment by dneimke** [comment]
OK, thanks @tillig.  We're a bit stuck in that we need to keep compat with the latest -unstable builds.  Any idea how much pain I'd be in for if I chose to fork Autofac and make those changes myself? :)

**Comment by tillig** [comment]
Couldn't say. I haven't had time to research it because, as mentioned, "the day job" has been eating my time. If I'm going to invest the time to switch, it's gotta be one-and-done, not "upgrade for the not-quite-ready-RC and also-be-compatible-with-beta-8" and so on. If you'd like to fork and give it a shot (or fork from develop and PR the result, _which would be welcome_), go for it.

Otherwise, I've gotta wait until -beta8 because of the semver problems that will be fixed in RC1.

**Comment by dneimke** [comment]
OK, thanks.  I'll take a deep breath, have a coffee and let you know how I decide to proceed :)

**Comment by dneimke** [comment]
I made the relevant changes to references and using statements and have published the updates here:

https://github.com/dneimke/Autofac/tree/develop

I think we will push a version to MyGet later today for anybody wishing to consume the latest version until an official release is made.  I'll update this with a link to the MyGet project when it's pushed up there.

**Comment by jruckert** [comment]
@dneimke packages have been pushed to 

https://www.myget.org/F/oconicsci/api/v3/index.json

**Comment by dneimke** [comment]
Cheers, thanks @jruckert ... love your work as always!  :)

**Comment by tillig** [comment]
Per [Damian Edwards on October 8](https://twitter.com/DamianEdwards/status/652230483305201664):

> FYI #aspnet5 beta8 is on day-2-day slip due to blocking issues. Fixes incoming but build-lag is ~2 days. I'll update when blocks are cleared

I have not seen any update since that tweet reflecting any change, so this is still on hold.

**Comment by tillig** [comment]
[It appears beta 8 is still blocked.](https://twitter.com/DamianEdwards/status/653628974371725312)

**Comment by alexmg** [comment]
Looks like the rename didn't make it into beta8.

https://www.nuget.org/packages/Microsoft.Framework.DependencyInjection

**Comment by jruckert** [comment]
Beta8 has not been published properly through to NuGet yet -> https://www.myget.org/gallery/aspnetvnext - master branch is still beta7

**Comment by alexmg** [comment]
Even on the aspnetvnext feed there is no beta8 package for `Microsoft.Extensions.DependencyInjection`. The only package is `1.0.0-rc1-15672`. That aligns with what was mentioned by @pranavkm in https://github.com/aspnet/DependencyInjection/issues/297

**Comment by jruckert** [comment]
https://www.myget.org/gallery/aspnetmaster is the one you need to look at for beta8.

**Comment by pranavkm** [comment]
Sorry if I wasn't super clear in the work item. We made the renames in RC1. Beta8 will have the old names.

**Comment by tillig** [comment]
Looks like beta 8 is live so we can cut a final beta 8 and take the rename for RC 1 in the develop branch.

**Comment by alexmg** [comment]
I did the final beta8 packages today but kept the integration package name (and assembly/namespace) as Autofac.Framework.DependencyInjection to avoid confusion. The release is tagged and has been pushed to master.

**Comment by tillig** [comment]
Working on the RC1 compatible stuff. Wish there was a command like `dnvm list --available` to see what runtime versions are in the feed.

**Comment by tillig** [comment]
Updated the `develop` branch to use DNX RC1 with `Microsoft.Extensions.DependencyInjection 1.0.0-rc1-*`. This includes a rename to `Autofac.Extensions.DependencyInjection` (not an awesome name, but not like I can think of too much better; at least it's consistent). This includes getting the updated specification tests which appears to remove the `[Fact]` attribute from the `ServiceInstanceCanBeResolved` test.

**The updated package can be found on [the Autofac MyGet feed](https://www.myget.org/F/autofac/api/v2) as `Autofac.Extensions.DependencyInjection 4.0.0-rc1-162`.**

I'm going to close this since the compatibility update is complete; if it turns out there's something else wrong, feel free to leave a note and we'll check it out.

**Comment by dneimke** [comment]
Thanks @tillig
